Automattic/pocket-casts-ios
# Make Pocket Casts compatible with the favourite (star) functionality introduced in iOS 17.1

Fixes #1194

Make Pocket Casts compatible with the favourite functionality introduced in iOS 17.1.

## To test
1. Disable _Extra Playback Actions_ in settings.
2. Play an episode. Note that star icon is _not present_ on native iOS Now Playing widget on lock screen or control centre.
3. Enable _Extra Playback Actions_ in settings.
4. Play an episode. Note that star icon _is present_ on Now Playing widget on lock screen or control centre.
5. Tap star icon on widget. Star should animate and remain filled.
6. Go to episode in app. Ensure star is applied.
7. Go to full screen player. Add Star to shelf if not present. Note that episode is starred in shelf.
8. Click star on shelf. Go back to Now Playing widgets and ensure that star is not filled.
9. Disable _Extra Playback Actions_ in settings. Note that star icon is no longer visible on Now Playing widgets.

## Checklist
